### PR TITLE
Add basic documentation.

### LINKED ISSUE #608: Add small documentation and transiton guide from libzim6 to libzim7

### PR DESCRIPTION
**Pull request by mgautierfr**
- Basic usage. Mainly taken from https://wiki.openzim.org/wiki/Libzim with (a lot of) adaptation.
- Transition guide from libzim6 to libzim7.

This is a basic documentation and doesn't cover all features of libzim.

Fix #608 

Generated doc can be seen here : https://libzim--611.org.readthedocs.build/en/611/

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+Errors are handled with exceptions. When something goes wrong, libzim throws an error,
+which is always derived from std::exception.
+
+All classes are defined in the namespace zim.
+Copying is allowed and tried to make as cheap as possible.
+The reading part of the libzim is most of the time thread safe.
+Searching and creating part are not. You have to serialize access to the class yourself.
+
+The main class, which accesses a archive is |Archive|.
+It has actually a reference to a implementation, so that copies of the class just references the same file.
```
Comment (maneeshpm) [inline docs/usage.rst:19]: reference to *an* implementation

**Review by maneeshpm — CHANGES_REQUESTED** [review]
Just some minor typo errors and a few suggestions. Good one!

**Review by rgaudin — APPROVED** [review]
Great addition

**Review by kelson42 — APPROVED** [review]

**Review by veloman-yunkan — APPROVED** [review]
Fixing typos & grammar and improving some pieces of the (new) documentation will unnecessarily delay merging this PR. It is a good start, we can merge it as is and then have someone refine it.

**Review by maneeshpm — APPROVED** [review]
I agree with @veloman-yunkan, we can refine it later.

File: docs/6to7.rst [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    search.set_query("bar");
+    search.set_range(10, 30);
+    for (auto it =search.begin(); it!=search.end(); it++)
+    {
+      std::cout << "Found result " << it.get_url() << std::endl;
+    }
+
+In libzim7 you search starting from a |Searcher|.
+
+ .. code-block:: c++
```
[the remark this answers is not in the document]
Comment (mgautierfr) [reply docs/6to7.rst:173]: I've removed all includes for consistency.

File: docs/6to7.rst [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    }
+
+If the zim file had no suggestion database, the suggestion search was made on full text database
+(whit variable results).
+
+In libzim7 you do suggestion using |SuggestionSearcher| API :
+
+ .. code-block:: c++
+
+    #include <zim/archive.h>
```
[the remark this answers is not in the document]
Comment (mgautierfr) [reply docs/6to7.rst]: I've removed all includes for consistency.

**Comment by mgautierfr** [comment]
Rebased on master and I've fixed the few typos.
Feel free to create PR to improve the documentation.

Deb packaging CI is failing but nothing related to this PR. I'm merging.

**Pull request merged**